keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -697,43 +699,51 @@ generatedFrom:
   apiGroup: dra.example.com
   uid: foobar-uid
 
-vendorParameters:
-  # A vendor can put any kind of object here to pass the configuration
-  # parameters down to the kubelet plugin. In this case, the vendor
-  # driver simply copied the entire CR. It could also be some
-  # separate, smaller configuration type.
-  #
-  # Beware that ResourceClaimParameters have separate RBAC rules than
-  # the vendor CRD, so information included here may get visible
-  # to more users than the original CRD. Both objects are in the same
-  # namespace.
-  kind: CardParameters
-  apiVersion: dra.example.com/v1alpha1
-  metadata:
-    name: my-parameters
-    namespace: user-namespace
-    uid: foobar-uid
-  ...
-  spec:
-    minimumRuntimeVersion: v12.0.0
-    minimumMemory: 32Gi
-    sharing:
-      strategy: TimeSliced
-
-driverRequests:
-- driverName: cards.dra.example.com
-  requests:
-  # Each entry here is a request for one resource.
-  - namedresources:
+requests:
+- vendorParameters:
+    # A vendor can put any kind of object here to pass the configuration
+    # parameters down to the kubelet plugin. In this case, the vendor
+    # driver simply copied the entire CR. It could also be some
+    # separate, smaller configuration type.
+    #
+    # Beware that ResourceClaimParameters have separate RBAC rules than
+    # the vendor CRD, so information included here may get visible
+    # to more users than the original CRD. Both objects are in the same
+    # namespace.
+    kind: CardParameters
+    apiVersion: dra.example.com/v1alpha1
+    metadata:
+      name: my-parameters
+      namespace: user-namespace
+      uid: foobar-uid
+    ...
+    spec:
+      minimumRuntimeVersion: v12.0.0
+      minimumMemory: 32Gi
+      sharing:
+        strategy: TimeSliced
+  # Each entry here is a request for one resource. Entries could have
+  # their own vendor parameters (not shown here).
+  subRequests:
+  - namedResources:
+      driverName: cards.dra.example.com
       # Selectors are CEL expressions with access to the attributes of the named resource
       # that is being checked for a match.
       selector: |-
+        # Attribute names can be fully qualified or shortened when they
+        # have the driver name as suffix.
+        attributes.string.has("UUID.cards.dra.example.com") &&
```

Comment:
It's better to add a "type" field in the ResourceSlice and check for that here.

The meaning of that "type" then is "all instances of a certain type are guaranteed to have certain attributes set".